### PR TITLE
Add new device handler for the BeNext Tag Reader

### PR DESCRIPTION
**Pull request by Klagopsalmer**
Hello,

I have added the support for the BeNext Tag Reader: http://www.benext.eu/en/products/tag-reader/

Commit log:
- Add UserCodeValue to store the data from the tag reader
- Create new ParameterEvent to notify the user when an unknown tag is read
- Create new command to allow the user to save tag code

It's my first contribution to an open source project, feel free to tell me if I have made mistakes.

I also have coded an automation program to use with the tag reader, where should I put it ?

Regards
Alexandre Schnegg

### CODE REVIEW

File: MigFiles/SupportLibraries/ZWaveLib/Devices/Controller.cs [hunk 1]
```diff
@@ -668,7 +668,7 @@ private void CheckDeviceHandler(ZWaveNode node, ManufacturerSpecific manufacture
                 var typeList = GetTypesInNamespace(Assembly.GetExecutingAssembly(), "ZWaveLib.Devices.ProductHandlers.");
                 for (int i = 0; i < typeList.Count; i++)
                 {
-                    //Console.WriteLine(typelist[i].FullName);
+                    //Console.WriteLine(typeList[i].FullName);
```
Comment (genielabs) [inline MigFiles/SupportLibraries/ZWaveLib/Devices/Controller.cs]: Please revert Controller.cs as no real changes have been made here.

File: MigFiles/SupportLibraries/ZWaveLib/Devices/Values/UserCodeValue.cs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                userCode.tagCode = new byte[10];
+                for (int i = 0; i < 10; i++)
+                {
+                    userCode.tagCode[i] = message[11 + i];
+                }
+            }
+            return userCode;
+        }
+
+        public byte[] getMessage()
```
Comment (genielabs) [inline MigFiles/SupportLibraries/ZWaveLib/Devices/Values/UserCodeValue.cs]: Use Camel Casing for methods:
https://msdn.microsoft.com/en-us/library/4df752aw%28v=vs.71%29.aspx

File: MigFiles/SupportLibraries/ZWaveLib/Devices/Values/UserCodeValue.cs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            ArrayList tempMessage=new ArrayList();
+            tempMessage.Add((byte)CommandClass.UserCode);
+            tempMessage.Add((byte)Command.UserCodeSet);
+            tempMessage.Add(userId);
+            tempMessage.Add(userIdStatus);
+            tempMessage.AddRange(tagCode);
+            return (byte[])tempMessage.ToArray(typeof(byte));
+        }
+
+        public string tagCodeToHexString()
```
Comment (genielabs) [inline MigFiles/SupportLibraries/ZWaveLib/Devices/Values/UserCodeValue.cs]: Use Camel Casing for methods:
https://msdn.microsoft.com/en-us/library/4df752aw%28v=vs.71%29.aspx

**Comment by genielabs** [comment]
Hi @Klagopsalmer, thanks for your contribution. I will merge as soon as you made those little changes to the code. For the automation program, export it and attach the hgx here or in a new issue so I will include it in the repo. Actually the procedure to add automation programs is tricky, so I'm doing this manually.

**Comment by Klagopsalmer** [comment]
Hello,

I have made the changes you asked for. 

The automation program is here : https://www.dropbox.com/s/xa86erkc7nb4aiq/1000-Tag_Reader.hgx?dl=0

Regards
Alexandre Schnegg

**Comment by genielabs** [comment]
Hi @Klagopsalmer , I had to manually add your code to the new ZWaveLib.
Also included your automation program with id 504.
The event that was called "UserCode.Report" has been renamed to "EntryControl.UserCode" both in MIG ZWave.cs and in the automation program.
Please check that everything is working as it should.

**Pull request closed without merging**